### PR TITLE
[Clang] Don't crash if input file is not a module.

### LINKED ISSUE #98365: [clang] Segfault with `-module-file-info`
If clang is invoked with `-module-file-info` and the file is of incorrect type then clang segfaults.

There is assert in the code
`assert(isCurrentFileAST() && "dumping non-AST?");`
Should this behavior be fixed or left unchanged?

CC: @AaronBallman @cor3ntin @Endilll

### PR DESCRIPTION
**Pull request by chestnykh**
Currently clang crashes with `-module-file-info` and input file which is not a module
Emit error instead of segfaulting.
Fix #98365

### CODE REVIEW

**Comment by chestnykh** [comment]
CC: @ChuanqiXu9 @shafik

File: clang/include/clang/Basic/DiagnosticFrontendKinds.td [hunk 1]
```diff
@@ -226,6 +226,7 @@ def err_module_map_not_found : Error<"module map file '%0' not found">,
 def err_missing_module_name : Error<
   "no module name provided; specify one with -fmodule-name=">,
   DefaultFatal;
+def err_file_is_not_module : Error<"file '%0' is not a module">, DefaultFatal;
```
Comment (ChuanqiXu9) [inline clang/include/clang/Basic/DiagnosticFrontendKinds.td:229]:
```suggestion
def err_file_is_not_module : Error<"file '%0' is not a module file">, DefaultFatal;
```

**Review by ChuanqiXu9 — APPROVED** [review]
LGTM

File: clang/include/clang/Basic/DiagnosticFrontendKinds.td (hunk 1 above)
Comment (chestnykh) [reply clang/include/clang/Basic/DiagnosticFrontendKinds.td]: Done. Extended the description

Comment (chestnykh) [reply clang/include/clang/Basic/DiagnosticFrontendKinds.td]: @ChuanqiXu9 FYI i don't have commit access so when PR is ready please commit it

Comment (ChuanqiXu9) [reply clang/include/clang/Basic/DiagnosticFrontendKinds.td]: OK, I'll commit it.

**Review by shafik — COMMENTED** [review]
Thank you for this fix!

**Comment by shafik** [comment]
I don't think the build error is related to this change.

**Pull request merged**